src/metabase/automagic_dashboards/core.clj

```diff
@@ -471,15 +477,25 @@
 
 (defn- make-context
   [root rule]
-  (let [source-table  (:source-table root)
-        tables        (concat [source-table] (linked-tables source-table))
-        table->fields (comp (->> (db/select Field
-                                   :table_id        [:in (map :id tables)]
-                                   :visibility_type "normal")
-                                 field/with-targets
-                                 (group-by :table_id))
-                            :id)]
-    (as-> {:source-table (assoc source-table :fields (table->fields source-table))
+  (let [source        (:source root)
+        tables        (concat [source] (linked-tables source))
+        table->fields (if (instance? (type Table) source)
+                        (comp (->> (db/select Field
+                                              :table_id        [:in (map :id tables)]
```

Comment:
Need to check that `tables` is not empty here

Reply:
It can't be. Source will always be non-nil. I'll add a :pre though